src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx

```diff
@@ -124,15 +172,24 @@ const RuleEditor = createReactClass({
     this.props.api.request(endpoint, {
       method: isNew ? 'POST' : 'PUT',
       data,
-      success: resp => {
-        this.setState({error: null, loading: false, rule: resp});
-        // Redirect to correct ID if /new
-        if (isNew) {
-          browserHistory.replace(
-            recreateRoute(`${resp.id}/`, {...this.props, stepBack: -1})
-          );
+      success: (resp, _, jqXHR) => {
+        // if we get a 202 back it means that we have an async task
+        // running to lookup and verfity the channel id for Slack.
+        if (jqXHR.status === 202) {
+          const rule = {...this.state.rule};
+          this.setState({error: null, loading: true, rule, uuid: resp.uuid});
```

Comment:
@MeredithAnya confused why we setting rule back to the original value here.

Reply:
hmmm don't know why I thought i needed this - think we can just take this out